cmd/cloud-controller-manager/app/options/options.go

```diff
@@ -131,24 +131,27 @@ func NewDefaultComponentConfig(insecurePort int32) (componentconfig.CloudControl
 	return internal, nil
 }
 
-// AddFlags adds flags for a specific ExternalCMServer to the specified FlagSet
-func (o *CloudControllerManagerOptions) AddFlags(fs *pflag.FlagSet) {
-	o.CloudProvider.AddFlags(fs)
-	o.Debugging.AddFlags(fs)
-	o.GenericComponent.AddFlags(fs)
-	o.KubeCloudShared.AddFlags(fs)
-	o.ServiceController.AddFlags(fs)
+// Flags returns flags for a specific APIServer by section name
+func (o *CloudControllerManagerOptions) Flags() (fss apiserverflag.NamedFlagSets) {
+	o.CloudProvider.AddFlags(fss.FlagSet("cloud provider"))
+	o.Debugging.AddFlags(fss.FlagSet("debugging"))
+	o.GenericComponent.AddFlags(fss.FlagSet("generic"))
+	o.KubeCloudShared.AddFlags(fss.FlagSet("generic"))
```

Comment:
Wondering if we want to categorize this as "cloud provider" as well. Not blocking though, will think about it and send open another PR if I think it should :)

Reply:
which one? `KubeCloudShared`? i think should not:grinning: